syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "custom_types.proto";

service TransactionFeeService {
  rpc GetAll (GetAllRequest) returns (GetAllTransactionFeesResponse);
  rpc GetById (GetTransactionFeeByIdRequest) returns (TransactionFeeModel);
}

message TransactionFeeModel {
  int32 id = 1;
  optional int32 transaction_id = 2;
  optional int32 fee_id = 3;
  optional string amount = 4;
}

message GetAllTransactionFeesResponse {
  repeated TransactionFeeModel transaction_fees = 1;
}

message GetTransactionFeeByIdRequest {
  int32 id = 1;
}
